Detections/MultipleDataSources/AADHostLoginCorrelation.yaml

```diff
@@ -0,0 +1,51 @@
+id:  8ee967a2-a645-4832-85f4-72b635bcb3a6
+name: IP with multiple failed AAD login attempts and successful remote login to host.
+description: |
+	This query looks for IP addresses with multiple failed logins to Azure AD and
+	within the same timeframe a successful login to a host via a remote protocol such as RDP or SSH.
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+     - SigninLogs
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvents
+  - connectorId: Syslog
+    dataTypes:
+      - Syslog 
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  
+query: |
+
+	let signin_threshold = 5; 
+	let suspicious_signins =
+	SigninLogs
+	| where TimeGenerated >= ago(1d)
+	| where ResultType !in ("0", "50125", "50140")
+	| summarize count() by IPAddress
+	| where count_ > signin_threshold
+	| summarize makelist(IPAddress);
+	let linux_logons =
+	Syslog
+	| where TimeGenerated >= ago(1d)
+	| where Facility contains "auth" and ProcessName != "sudo"
+	| where SyslogMessage has "Accepted"
+	| where SyslogMessage in (suspicious_signins)
```

Comment:
Is the IPAddress the whole syslogmessage? There's a gnarly way look for items in a list inside something else ISTR but you have to join each ip address entry to you target data and do message contains(ip)

Reply:
or use a regex to extract the ip

Reply:
Thanks, yes meant to use has_any for the syslog look up, have fixed this now.

Reply:
Had to extract the IP in the end to support the entity mapping in the new standards so fell back to in.